src/sys/socket/ffi.rs

```diff
@@ -33,9 +41,9 @@ pub struct msghdr<'a> {
     pub msg_name: *const c_void,
```

Comment:
This struct should be replaced with the FFI types declared in libc, but that doesn't need to be done for this PR. Please add a FIXME stating that.